### PR TITLE
Improve block tracker computation time

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Improved block tracker computation time, when received headers are behind final or if registered handlers lists which should be notified on new headers received are empty

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: process/mock/blockNotifierHandlerMock.go [hunk 1]
```diff
@@ -6,8 +6,9 @@ import (
 
 // BlockNotifierHandlerMock -
 type BlockNotifierHandlerMock struct {
```
Comment (bogdan-rosianu) [inline process/mock/blockNotifierHandlerMock.go:8]: could be renamed to stub

File: process/track/blockProcessor_test.go [hunk 2]
```diff
@@ -174,10 +186,15 @@ func TestProcessReceivedHeader_ShouldWorkWhenHeaderIsFromSelfShard(t *testing.T)
 			return nil, nil, nil, nil
 		},
 	}
+	blockProcessorArguments.SelfNotarizer = &mock.BlockNotarizerHandlerMock{
+		GetLastNotarizedHeaderCalled: func(shardID uint32) (data.HeaderHandler, []byte, error) {
+			return &block2.Header{}, nil, nil
```
Comment (bogdan-rosianu) [inline process/track/blockProcessor_test.go:191]: rename package from `block2` to something else

**Review by bogdan-rosianu — APPROVED** [review]

File: process/track/blockProcessor_test.go (hunk 2 above)
Comment (SebastianMarian) [reply process/track/blockProcessor_test.go:191]: done

File: process/mock/blockNotifierHandlerMock.go (hunk 1 above)
Comment (SebastianMarian) [reply process/mock/blockNotifierHandlerMock.go]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by sasurobert — DISMISSED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**